library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -444,7 +444,7 @@ private NetworkState checkSizeAllowedForNetwork(int networkType) {
      */
     public boolean isReadyToDownload() {
         synchronized (this) {
-            return isDownloadManagerReadyToDownload() && isClientReadyToDownload();
+            return isClientReadyToDownload() && isDownloadManagerReadyToDownload();
```

Comment:
I think this is a bit of perf improvement. If the first one is `false` then the second won't be checked as it is an `&&`. If `isClientReadyToDownload()` takes less to execute than the other one and it is more likely to return `false` then I think we have a little bit of improvement.